coordinator/history/history.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package history
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"hash"
+
+	"github.com/spf13/afero"
+)
+
+const (
+	hashSize = 32 // byte, History.hashFun().Size()
+	histPath = "/mnt/state/history"
+)
+
+// History is the history of the Coordinator.
+type History struct {
+	store      store
+	hashFun    func() hash.Hash
+	signingKey *ecdsa.PrivateKey
+}
+
+// New creates a new History with the given signing key.
+func New() *History {
+	osFS := afero.NewOsFs()
+	if err := osFS.MkdirAll(histPath, 0o755); err != nil {
+		panic(fmt.Errorf("creating history directory: %w", err))
+	}
+	h := &History{
+		store:   newPVStore(&afero.Afero{Fs: afero.NewBasePathFs(osFS, histPath)}),
+		hashFun: sha256.New,
+	}
+	if hashSize != h.hashFun().Size() {
+		panic("mismatch between hashSize and hash function size")
```

Comment:
Out of curiosity, what is the tradeoff between panic and erroring here? Is having a New function that can return an error discouraged practice?

Reply:
There's no meaningful way to handle this error - the package is just broken if this assertion does not hold. With that in mind, I appreciate that panicking gives us an error-return-free `New` function.

Reply:
It is discouraged to panic, that's right. Actually, as there is now a second function that might fail, which isn't a sanity check, we should return the error (sorry @burgerdev :disappointed: , I like constructors without error as well. )